src/main/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannel.java

```diff
@@ -1303,4 +1317,45 @@ public String toString() {
       return "[" + this.name() + "]";
     }
   }
+
+  /**
+   * Registers all the Metrics inside the metricRegistry.
+   *
+   * @param channelName channelName
+   * @param metricsJmxReporter wrapper class for registering all metrics related to above connector
+   *     and channel
+   */
+  public void registerChannelJMXMetrics(
```

Comment:
Could we add latency related metrics similar to https://docs.snowflake.com/en/user-guide/kafka-connector-monitor#category-latencies?

Reply:
yes, i'm planning to add that in a follow up pr. wanted to keep this one small to just enable JMX, and then add latencies and buffer thresholds later